bin/drop_counts.py

```diff
@@ -4,134 +4,128 @@
 import re
 from collections import OrderedDict
 from pathlib import Path
-
+import sys
 import pandas as pd
+from typing import Set, Dict
 
-translator = {
+SCRIPT_VERSION = "v1.0"
+TRANSLATOR = {
     "unstranded": 1,
     "forward": 2,
     "reverse": 3,
 }
 
-
-def get_non_std_genes(gtf: Path) -> set[str]:
+STANDARD_CHROMOSOMES = [
+    "chr1",
+    "chr2",
+    "chr3",
+    "chr4",
+    "chr5",
+    "chr6",
+    "chr7",
+    "chr8",
+    "chr9",
+    "chr10",
+    "chr11",
+    "chr12",
+    "chr13",
+    "chr14",
+    "chr15",
+    "chr16",
+    "chr17",
+    "chr18",
+    "chr19",
+    "chr20",
+    "chr21",
+    "chr22",
+    "chrX",
+    "chrY",
+    "chrM",
+]
+
+
+def get_non_std_genes(gtf: Path) -> Set[str]:
     """Create list of genes not belonging to chr1-21 or chrM"""
-
-    std_chrs = [
-        "chr1",
-        "chr2",
-        "chr3",
-        "chr4",
-        "chr5",
-        "chr6",
-        "chr7",
-        "chr8",
-        "chr9",
-        "chr10",
-        "chr11",
-        "chr12",
-        "chr13",
-        "chr14",
-        "chr15",
-        "chr16",
-        "chr17",
-        "chr18",
-        "chr19",
-        "chr20",
-        "chr21",
-        "chr22",
-        "chrX",
-        "chrY",
-        "chrM",
-    ]
-
     gene_id_regex = re.compile('gene_id "(.+?)"')
     genes_to_exclude = []
-
     with open(gtf, "r") as gtf_file:
         for line in gtf_file:
             if line.startswith("#"):
                 continue
-
-            if line.split()[0] in std_chrs:
+            if line.split()[0] in STANDARD_CHROMOSOMES:
                 continue
-
             gene_id = re.search(gene_id_regex, line)
             genes_to_exclude.append(gene_id.group(1))
-
     return set(genes_to_exclude)
 
 
-def read_star_gene_cnts(sample: str, star: Path, strandedness: str) -> dict:
+def read_star_gene_counts(sample: str, star: Path, strandedness: str) -> Dict:
     """Read gene count file(s) from STAR output to return sample_ids."""
     sample_ids = {}
     gene_ids = {}
     with open(star) as in_tab:
         for line in in_tab:
             if not line.startswith("N_"):
                 gene_id = line.split()[0]
-                strand = translator[strandedness]
-                counts = line.split()[strand]
-                gene_ids[gene_id] = int(counts)
+                strand = TRANSLATOR[strandedness]
+                counts = int(line.split()[strand])
+                gene_ids[gene_id] = counts
```

Comment:
let's not call split() twice. Define a variable and use it to retrieve the values you need

Reply:
Done